src/plugins/janus_sip.c

```diff
@@ -1771,6 +1772,127 @@ static void janus_sip_sofia_logger(void *stream, char const *fmt, va_list ap) {
 	}
 }
 
+/* Sofia SIP logger function: when the Event Handlers mechanism is enabled,
+ * we use this to intercept SIP messages sent by the stack (received
+ * messages are more easily recoverable in janus_sip_sofia_callback) */
+static void janus_sip_sofia_logger_single(void *stream, char const *fmt, va_list ap) {
+	if(!fmt)
+		return;
+	char line[255];
+	char message[16000];
+	char sofia_log_buf[16000];
+	char call_id_hdr[255];
+	sofia_log_buf[0] = '\0';
+	call_id_hdr[0] = '\0';
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-nonliteral"
+#ifndef __clang__
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wsuggest-attribute=format"
+#endif
+	g_vsnprintf(line, sizeof(message), fmt, ap);
+#ifndef __clang__
+#pragma GCC diagnostic pop
+#endif
+#pragma GCC diagnostic pop
+	gchar **parts = g_strsplit(message, "\n", -1);
+	if (parts) {
+		int index = 0;
+		char *line = NULL, *cr = NULL;
+		while((line = parts[index]) != NULL) {
+			cr = strchr(line, '\r');
+			if(cr != NULL)
+				*cr = '\0';
+			if(*line == '\0') {
+				if(cr != NULL)
+					*cr = '\r';
+				index++;
+				continue;
+			}
+
+			/* First line of the log indicates if we send or receive a message. */
+			/* We are only interested in the sent ones */
+			if(index==0) {
+				if(strstr(line, "send ") == line) {
+					int length = atoi(line+5);
+					JANUS_LOG(LOG_HUGE, "Intercepting message (%d bytes)\n", length);
+					index++;
+					continue;
+				}
+				g_strfreev(parts);
+				return ;
+			}
+
+			if(strlen(line) == 1) {
```

Comment:
As explained in #3059, this should use `strnlen` instead of `strlen`, to optimize performance. Since you're interested in checking if the size is exactly 1, I guess the right approach is passing at least 2, e.g.:

    if(strnlen(line, 1 + 1) == 1) {

(but @tmatth may correct me if I'm wrong).